client/js/widgets/button.js

```diff
@@ -514,6 +514,27 @@ export class Button extends Widget {
 
         const dom = document.createElement('div');
 
+        if(field.applyVariables) {
+          if(Array.isArray(field.applyVariables)) {
+            for(const v of field.applyVariables) {
+              if(v.parameter && v.variable) {
+                field[v.parameter] = (v.index === undefined) ? variables[v.variable] : variables[v.variable][v.index];
+              } else if(v.parameter && v.template) {
+                field[v.parameter] = v.template.replace(/\{([^}]+)\}/g, function(i, key) {
+                  return (variables[key] === undefined) ? "" : variables[key];
+                });
+              } else if(v.parameter && v.property) {
+                let w = widgets.has(v.widget) ? widgets.get(v.widget) : this;
+                field[v.parameter] = (w.p(v.property) === undefined) ? null : w.p(v.property);
+              } else {
+                problems.push('Entry in parameter applyVariables does not contain "parameter" together with "variable", "property", or "template".');
```

Comment:
I don't think there is a problems array in this context.

Reply:
Take a look at my changes in #217 (which is pointed at this branch). I replicated the test in its pr-test room.